src/System.Management.Automation/utils/Encoding.cs

```diff
@@ -0,0 +1,399 @@
+/********************************************************************++
+Copyright (c) Microsoft Corporation.  All rights reserved.
+--********************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Management.Automation;
+using System.Management.Automation.Provider;
+
+namespace Microsoft.PowerShell
+{
+    /// <summary>
+    /// The list of available file encodings
+    /// </summary>
+    public enum FileEncoding
+    {
+        /// <summary>
+        /// No encoding, or unset.
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// Unicode encoding.
+        /// </summary>
+        String,
+
+        /// <summary>
+        /// Unicode encoding.
+        /// </summary>
+        Unicode,
+
+        /// <summary>
+        /// Byte encoding.
+        /// </summary>
+        Byte,
+
+        /// <summary>
+        /// Big Endian Unicode encoding.
+        /// </summary>
+        BigEndianUnicode,
+
+        /// <summary>
+        /// Backward compatibility - UTF8 encoding without BOM
+        /// </summary>
+        UTF8,
+
+        /// <summary>
+        /// UTF8 encoding which includes BOM.
+        /// </summary>
+        UTF8BOM,
+
+        /// <summary>
+        /// UTF8 encoding without BOM.
+        /// </summary>
+        UTF8NoBOM,
+
+        /// <summary>
+        /// UTF7 encoding.
+        /// </summary>
+        UTF7,
+
+        /// <summary>
+        /// UTF32 encoding.
+        /// </summary>
+        UTF32,
+
+        /// <summary>
+        /// ASCII encoding.
+        /// </summary>
+        Ascii,
+
+        /// <summary>
+        /// Default encoding.
+        /// </summary>
+        Default,
+
+        /// <summary>
+        /// OEM encoding.
+        /// </summary>
+        Oem,
+
+        /// <summary>
+        /// Big Endian UTF32 encoding
+        /// </summary>
+        BigEndianUTF32,
+
+        /// <summary>
+        /// Windows legacy encoding. This requires a cmdlet object to resolve.
+        /// </summary>
+        WindowsLegacy,
+    }
+
+    /// <summary>
+    /// the helper class for determining encodings for PowerShell
+    /// </summary>
+    public static class PowerShellEncoding
+    {
+
+        /// <summary>
+        /// Return the default PowerShell encoding which is UTF8 without a BOM.
+        /// There is no distinction between platforms
+        /// </summary>
+        public static Encoding GetDefaultEncoding()
+        {
+            return utf8NoBom;
+        }
+
+        /// <summary>
+        /// translate a FileEncoding to an actual System.Text.Encoding
+        /// <param name="TextEncoding">The enum value</param>
+        /// <returns>System.Text.Encoding</returns>
+        /// </summary>
+        public static Encoding GetEncoding(FileEncoding TextEncoding)
```

Comment:
Shouldn't I be able to use this api to get an Encoding for a file I've written using `FileEncoding`? That will work for everything but `WindowsLegacy`, so I would say it's better to throw.